torch/distributed/utils.py

```diff
@@ -16,20 +18,26 @@ def _recursive_to(inputs, target_gpu, use_side_stream_for_tensor_copies):
     """
 
     def to_map(obj):
+        ddp_device_type = _get_available_device_type()
+        if ddp_device_type is not None:
```

Comment:
This method is called multiple times during initialization, we should move this to be done only once on the top level of this module? IE:

```python
@functools.cache
def _get_device_impl():
    ddp_device_type = _get_available_device_type()
     if ddp_device_type is not None:
         ddp_torch_module = importlib.import_module("torch." + ddp_device_type)
     else:
          raise ImportError("Failed to import the gpu device module of pytorch (torch.cuda or torch.xpu).")
    return (ddp_device_type, ddp_torch_module)
```